### PR TITLE
FF115 URLSearchParams.has() and delete() value arg

### LINKED ISSUE #26604: Document value argument to URLSearchParams's has() and delete()
This was the most-requested feature in the URL Standard backlog.

Standard: https://github.com/whatwg/url/pull/735.

WPT: https://github.com/web-platform-tests/wpt/pull/39865.

### PR DESCRIPTION
**Pull request by hamishwillee**
FF115 `URLSearchParams.has()` and `URLSearchParams.delete()` have a new optional `value` argument to go with the parameter name.
Without the value the methods match on all/any parameter that has the specified name. With the `value` argument they methods match on a parameter that has both the matching name and value.

This updates the documentation for the two methods.

Other docs work for this can be tracked in #27172

Fix #26604

### CODE REVIEW

File: files/en-us/web/api/urlsearchparams/delete/index.md [hunk 1]
```diff
@@ -8,37 +8,91 @@ browser-compat: api.URLSearchParams.delete
...
+log(`Query string (before):\t ${params}`);
+params.delete("foo", "1");
+log(`Query string (after):\t ${params}`);
 ```
 
+All parameters that match both the parameter `name` and `value` should be deleted (there is no reason to specify two parameters with the same name and value as shown above).
+
+{{EmbedLiveSample('Delete parameters with specified name and value', '100%', '50')}}
+
+If your browser supports the `value` option, the "after" string should be `bar=2&foo=3`.
```
Comment (hamishwillee) [inline files/en-us/web/api/urlsearchparams/delete/index.md:93]: Note, I don't know of any way to check whether this feature is supported in a browser, other than running a test case and seeing if you get an expected value. Decided not to comment.

**Review by zloirock — COMMENTED** [review]
`core-js` has a polyfill for this. Could you add [a link](https://github.com/zloirock/core-js#url-and-urlsearchparams)?

**Comment by hamishwillee** [comment]
@zloirock Thanks. I've added core.js links to the items I'm touching here - not to "everything"

**Comment by hamishwillee** [comment]
@Elchi3 When you have a moment, can I please get this reviewed. No urgency.

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**